js/ui/audio_worker.ts

```diff
@@ -0,0 +1,47 @@
+
+declare global {
+    interface AudioWorkletProcessor {
+        readonly port: MessagePort;
+        process(inputs: Float32Array[][], outputs: Float32Array[][], parameters: Map<string, Float32Array>): void;
+    }
+
+    const AudioWorkletProcessor: {
+        prototype: AudioWorkletProcessor;
+        new(options?: AudioWorkletNodeOptions): AudioWorkletProcessor;
+    };
+
+    function registerProcessor(name: string, ctor :{ new(): AudioWorkletProcessor; }): void
+}
+
+export class AppleAudioProcessor extends AudioWorkletProcessor {
+    private samples: Float32Array[] = []
+
+    constructor() {
+        super();
+        console.info('AppleAudioProcessor constructor');
+        this.port.onmessage = (ev: MessageEvent) => {
+            this.samples.push(ev.data);
+            if (this.samples.length > 256) {
+                this.samples.shift();
+            }
+        };
+    }
+
+    static get parameterDescriptors() {
+        return [];
+    }
+
+    process(_inputList: Float32Array[][], outputList: Float32Array[][], _parameters: Map<string, Float32Array>) {
+        const sample = this.samples.shift();
+        const output = outputList[0];
+        if (sample) {
+            for (let idx = 0; idx < sample.length; idx++) {
+                output[0][idx] = sample[idx];
+            }
+        }
+
+        return true;
```

Comment:
Maybe note that this forces the processor to stay alive because it doesn't have any inputs and, frequently, the outputs stay the same for long periods of time, which the browser might interpret as "inactivity" and shut it down.